lib/improver/spotdata_new/apply_lapse_rate.py

```diff
@@ -0,0 +1,206 @@
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""Apply temperature lapse rate adjustments to a spot data cube."""
+
+import numpy as np
+
+import iris
+from improver.spotdata_new.spot_extraction import SpotExtraction
+
+
+class SpotLapseRateAdjust:
+    """
+    Adjusts spot data temperatures by a lapse rate to better represent the
+    conditions at their altitude that may not be captured by the model
+    orography.
+    """
+
+    def __init__(self, neighbour_selection_method='nearest',
+                 grid_metadata_identifier='mosg'):
+        """
+        Keyword Args:
+            neighbour_selection_method (str):
+                The neighbour cube may contain one or several sets of grid
+                coordinates that match a spot site. These are determined by
+                the neighbour finding method employed. This keyword is used to
+                extract the desired set of coordinates from the neighbour cube.
+                The methods available will be all, or a subset, of the
+                following::
+
+                  - nearest
+                  - nearest_land
+                  - nearest_land_minimum_dz
+                  - nearest_minimum_dz
+
+                The method available in a neigbour cube will depend on the
+                options that were specified when it was created.
+            grid_metadata_identifier (str or None):
+                A string to search for in the input cube's attributes that
+                can be used to ensure that the cubes being used are for the
+                same model/grid. This test can be bypassed by setting this to
+                None.
+        """
+        self.neighbour_selection_method = neighbour_selection_method
+        self.grid_metadata_identifier = grid_metadata_identifier
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        return ('<SpotLapseRateAdjust: neighbour_selection_method: {}, '
+                'grid_metadata_identifier: {}>'.format(
+                    self.neighbour_selection_method,
+                    self.grid_metadata_identifier))
+
+    def _get_attributes(self, cube):
+        """
+        Build dictionary of attributes that match the
+        self.grid_metadata_identifier.
+
+        Args:
+            cube (iris.cube.Cube):
+                A cube from which attributes partially matching the
+                self.grid_metadata_identifier will be returned.
+        Returns:
+            attributes (dict):
+                A dictionary of attributes partially matching
+                self.grid_metadata_identifier that were found on the input
+                cube.
+        """
+        attributes = cube.attributes
+        attributes = {k: v for (k, v) in attributes.items()
+                      if self.grid_metadata_identifier in k}
+        return attributes
+
+    def _compare_attributes(self, attributes, reference_attributes):
```

Comment:
Is there not an analogous function in cube_manipulation?  Is it possible to use that function here,  or to move some functionality from here into the more general utility to make it possible?

Reply:
(Feel free to move this function into a cube_manipulation utility and / or refactor the existing function to make it more concise / elegant if you would like.)

Reply:
:+1: